packages/contracts/utils/etherscan.ts

```diff
@@ -77,3 +82,417 @@ export const runTaskWithRetry = async (
     }
   }
 };
+
+export type AllDeployments = {[name: string]: Deployment};
+
+/**
+ * @returns All contracts with the name containing 'Proxy' from the deployments stored on disk
+ */
+export const fetchProxies = (deployments: AllDeployments) =>
+  Object.entries(deployments)
+    .map(([name, deploy]) => ({name, address: deploy.address.toLowerCase()}))
+    .filter(deployment => deployment.name.match(/Proxy$/g))
+    // Naming conventions is to call the base contract XXXProxy, but HH Deploy
+    // will suffix _Proxy (leading to XXXProxy_Proxy).
+    // This second filter removes that noise as they're the same contract anyway.
+    .filter(deployment => !deployment.name.match(/Proxy\_Proxy$/g));
+
+/**
+ * @returns All contracts with the name containing 'Implementation' from the deployments stored on disk
+ */
+export const fetchImplementations = (deployments: AllDeployments) =>
+  Object.entries(deployments)
+    .map(([name, deploy]) => ({name, address: deploy.address.toLowerCase()}))
+    .filter(deployment => deployment.name.match(/Implementation$/g));
+
+export type CollectedProxyWithImplementation = {
+  name: string;
+  proxy: {name: string; address: string};
+  implementation: {name: string; address: string} | undefined;
+};
```

Comment:
Perhaps we need better organzation for our types, perhaps in a different file...etc

Reply:
Happy to do this, as a personal preference I like to keep types only used once near the function but I will absolutely follow the conventions set by the team